samples/bluetooth/observer/src/main.c

```diff
@@ -41,13 +109,17 @@ void main(void)
 		printk("Bluetooth init failed (err %d)\n", err);
 		return;
 	}
-
 	printk("Bluetooth initialized\n");
 
+#if defined(CONFIG_BT_EXT_ADV)
+	bt_le_scan_cb_register(&scan_callbacks);
+	printk("Registered scan callbacks\n");
+#endif /* CONFIG_BT_EXT_ADV */
```

Comment:
Why are we only registering scan callbacks when extended advertising is enabled?

Reply:
I was thinking the same. We should retire the `device_found` callback and only use the `scan_callbacks` IMO.

Reply:
> It is outside the scope of this PR to retire device_found.

It is fine to keep as is for me. 

I expect the reason why we have 2 scan callbacks is to avoid breaking a stable API, but might make sense to do at some point.